precompiles/staking/staking.go

```diff
@@ -109,31 +132,40 @@ func (p Precompile) Run(evm *vm.EVM, caller common.Address, input []byte) (bz []
 		return p.redelegate(ctx, method, caller, args)
 	case UndelegateMethod:
 		return p.undelegate(ctx, method, caller, args)
+	case DelegationQuery:
+		return p.getDelegation(ctx, method, caller, args)
+	case StakingPoolQuery:
+		return p.getStakingPool(ctx, method, args)
+	case UnbondingDelegationQuery:
+		return p.getUnbondingDelegation(ctx, method, args)
+	case UnbondingDelegationByUnbondingIDQuery:
+		return p.getUnbondingDelegationByUnbondingId(ctx, method, args)
+	default:
+		return
 	}
-	return
 }
 
 func (p Precompile) delegate(ctx sdk.Context, method *abi.Method, caller common.Address, args []interface{}) ([]byte, error) {
 	pcommon.AssertArgsLength(args, 2)
 	delegator := p.evmKeeper.GetSeiAddressOrDefault(ctx, caller)
-	validatorBech32 := args[0].(string)
+	validatorBech32 := p.evmKeeper.GetSeiAddressOrDefault(ctx, args[0].(common.Address)).String()
```

Comment:
validator address is different from normal account address and doesn't have associated EVM address, so we can't change this line

Reply:
What's the length of these ? 

I changed these do `address` mainly because they were previously `string` which in solidity is really a dynamic bytes array. 

So if the validator addresses are 32 Bytes or shorter we can use a normal value type like `bytes32`.

Reply:
kind bump @codchen